controllers/applicationvolumereplication_controller.go

```diff
@@ -195,55 +195,193 @@ func (r *ApplicationVolumeReplicationReconciler) Reconcile(ctx context.Context,
+}
 
-		return ctrl.Result{Requeue: true}, nil
+func (r *ApplicationVolumeReplicationReconciler) ensureSubsHaveSamePlacementRule(
+	avr *rmn.ApplicationVolumeReplication, subscriptions *subv1.SubscriptionList) error {
+	if len(subscriptions.Items) == 0 {
+		return fmt.Errorf("empty subscriptionList for avr %s", avr.Name)
 	}
 
-	r.Log.Info(fmt.Sprintf("AVR (%+v)", avr))
+	for _, sub := range subscriptions.Items {
+		if sub.Spec.Placement == nil || sub.Spec.Placement.PlacementRef == nil {
+			return fmt.Errorf("invalid subscription placement object. SubName %s", sub.Name)
+		}
+
+		if !reflect.DeepEqual(avr.Spec.Placement, sub.Spec.Placement) {
+			return fmt.Errorf("mismatch between subscription placement and avr placement (subName %s)", sub.Name)
+		}
+	}
 
-	logger.Info(fmt.Sprintf("Processed %d/%d subscriptions. AVR Status Decisions %d. Requeue? %v",
-		len(a.instance.Status.Decisions), len(subscriptionList.Items), len(avr.Status.Decisions), requeue))
+	return nil
+}
 
-	done := !requeue
-	r.Callback(a.instance.Name, done)
+func (r *ApplicationVolumeReplicationReconciler) ensurePlRuleClustersInDRClusterPeers(
+	avr *rmn.ApplicationVolumeReplication) error {
+	clPeersRef := avr.Spec.DRClusterPeersRef
+	clusterPeers := &rmn.DRClusterPeers{}
 
-	return ctrl.Result{Requeue: requeue}, nil
+	if clPeersRef.Namespace == "" {
+		clPeersRef.Namespace = avr.Namespace
+	}
+
+	err := r.Client.Get(context.TODO(),
+		types.NamespacedName{Name: clPeersRef.Name, Namespace: clPeersRef.Namespace}, clusterPeers)
+	if err != nil {
+		return fmt.Errorf("failed to get cluster peers using %s/%s. Error (%w)", clPeersRef.Name, clPeersRef.Namespace, err)
+	}
+
+	if len(clusterPeers.Spec.ClusterNames) == 0 {
+		return fmt.Errorf("invalid DRClusterPeers configuration. Name %s", clusterPeers.Name)
+	}
+
+	plRef := avr.Spec.Placement.PlacementRef
+
+	if plRef.Namespace == "" {
+		plRef.Namespace = avr.Namespace
+	}
+
+	placementRule := &plrv1.PlacementRule{}
+
+	err = r.Client.Get(context.TODO(), types.NamespacedName{Name: plRef.Name, Namespace: plRef.Namespace}, placementRule)
+	if err != nil {
+		return fmt.Errorf("failed to get placementrule error: %w", err)
+	}
+
+	const requiredClusterCount = 1
+
+	clmap, err := r.getManagedClustersUsingPlacementRule(placementRule, requiredClusterCount)
+	if err != nil {
+		return fmt.Errorf("failed to get cluster map for placement %s error: %w", placementRule.Name, err)
+	}
+
+	for clName := range clmap {
+		found := r.findClusterInDRClusterPeers(clusterPeers, clName)
+
+		if !found {
+			return fmt.Errorf("mismatch between PlRule managed clusters and DRClusterPeers. "+
+				"Cluster name %s not found in DRClusterPeers %v", clName, clusterPeers.Spec.ClusterNames)
+		}
+	}
+
+	return nil
+}
+
+func (r *ApplicationVolumeReplicationReconciler) getManagedClustersUsingPlacementRule(
+	placementRule *plrv1.PlacementRule, requiredClusterCount int) (map[string]*spokeClusterV1.ManagedCluster, error) {
+	const requiredClusterReplicas = 1
```

Comment:
What is the difference between the variable @requiredClusterCount and @requiredClusterReplicas? 

I assume the former is the number of clusters expected in the placement rule and the latter is the number of replicas of the subscription to be deployed in the managed clusters. Is this understanding correct?

Also, instead of using these local variables set to 1 inside these 2 functions (ensurePlRuleClustersInDRClusterPeers and getManagedClustersUsingPlacementRule), can they be defined (and set) as separate constant variables (may be at the beginning of this file or may be applicationvolumereplication_types.go file)? Intention being, in future if we want to change the requirement for either number of clusters or number of replicas, we can modify them just at the place of their definition. It might also be helpful while reading the code about these constraints. Just a thought. It looks fine with how it is currently in this PR. Let me know your throughts.

Reply:
Your understanding of requiredClusterCount and requiredClusterReplicas is correct.

For your second comment; I am containing those variables where they are being used because only those functions need them as of now.  I don't see a need for being global. Also, I declared them as 'const' for good practice only.